includes/device-groups.inc.php

```diff
@@ -153,6 +152,11 @@ function GenGroupSQL($pattern, $search = '', $extra = 0)
     if (!empty($join)) {
         $join = '('.rtrim($join, '& ').') &&';
     }
+  //patch sql error adding extra devices table in query when it has 3 elements 
+    $pos=array_search('devices',$tables);
+    if (count($tables)==3) {
+    array_pop($tables);
```

Comment:
It would possibly be better to just call `array_unique($tables)` to remove any duplicate tables.

This current code removes the last table when there happens to be 3 tables.

Reply:
Hello,

I am open to suggestion, but being a beginner in writing code and I just like to debug. I just wanted to help. I can redo the commit with array_unique.

Reply:
You can also update this PR.  Currently the code doesn't work as you intend (in some cases).

I think just `$tables = array_unique($tables);` should be all you need to add.